### PR TITLE
ipn/ipnserver: close a small race in ipnserver, ~simplify code

### PR DESCRIPTION
**Pull request by bradfitz**
There was a small window in ipnserver after we assigned a LocalBackend
to the ipnserver's atomic but before we Start'ed it where our
initalization Start could conflict with API calls from the LocalAPI.

Simplify that a bit and lay out the rules in the docs.

Updates #12028

### CODE REVIEW

File: cmd/tailscaled/tailscaled.go [hunk 1]
```diff
@@ -464,7 +465,7 @@ func startIPNServer(ctx context.Context, logf logger.Logf, logID logid.PublicID,
 		debugMux.HandleFunc("/debug/ipn", srv.ServeHTMLStatus)
 	}
 	var lbErr syncs.AtomicValue[error]
-
+	runErrc := make(chan error, 1)
```
Comment (maisem) [inline cmd/tailscaled/tailscaled.go:468]: Write a comment here, took me a bit to understand it. (Though I was reading on mobile, might be easier to follow on big screen)

File: cmd/tailscaled/tailscaled.go [hunk 2]
```diff
@@ -480,7 +481,24 @@ func startIPNServer(ctx context.Context, logf logger.Logf, logID logid.PublicID,
 		lb, err := getLocalBackend(ctx, logf, logID, sys)
 		if err == nil {
 			logf("got LocalBackend in %v", time.Since(t0).Round(time.Millisecond))
-			srv.SetLocalBackend(lb)
+			// Now wait for Run to have start.
```
Comment (maisem) [inline cmd/tailscaled/tailscaled.go:484]: Words

**Review by maisem — APPROVED** [review]

File: ipn/ipnserver/server.go [hunk 3]
```diff
@@ -490,21 +490,6 @@ func (s *Server) SetLocalBackend(lb *ipnlocal.LocalBackend) {
...
-func (b *Server) startBackendIfNeeded() {
-	if !b.runCalled.Load() {
-		return
-	}
-	lb := b.lb.Load()
-	if lb == nil {
-		return
-	}
-	if lb.Prefs().Valid() {
-		b.startBackendOnce.Do(func() {
```
Comment (maisem) [inline ipn/ipnserver/server.go:502]: Staticcheck complaining about this being unused now

File: cmd/tailscaled/tailscaled.go [hunk 4]
```diff
@@ -480,6 +481,15 @@ func startIPNServer(ctx context.Context, logf logger.Logf, logID logid.PublicID,
 		lb, err := getLocalBackend(ctx, logf, logID, sys)
 		if err == nil {
 			logf("got LocalBackend in %v", time.Since(t0).Round(time.Millisecond))
+			if lb.Prefs().Valid() {
+				if err := lb.Start(ipn.Options{}); err != nil {
```
Comment (maisem) [inline cmd/tailscaled/tailscaled.go:485]: line 503 doesn't block for this goroutine to end right?

Comment (maisem) [reply cmd/tailscaled/tailscaled.go:485]: so then 504 would never run?

Comment (maisem) [reply cmd/tailscaled/tailscaled.go:485]: oh wait Run never returns. nvm

Comment (bradfitz) [reply cmd/tailscaled/tailscaled.go:485]: line 501 blocks until the cancel after the error store:

```
					lbErr.Store(err)
					cancel()
```

So we store the error, then cancel, unblocking Run, then Run checks that atomic error.

**Review by maisem — APPROVED** [review]

**Pull request merged**